lib/dao.js

```diff
@@ -840,6 +840,19 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId) {
+    this.findById(id, function(err, instance) {
+      if (err || !instance) {
+        var msg = g.f('Cannot find unknown {{id}} "%s". ', id);
+        var error = new Error(msg);
+        error.statusCode = error.status = 404;
+        return cb(error);
+      }
+    });
+    return Model.replaceById(id, data, options, cb);
```

Comment:
I think `return Model.replaceById(id, data, options, cb);` this line should move into the callback function of `findById`?

```
this.findById(id, function(err, instance) {
  if (err) {
    // handle error
  } else {
    return Model.replaceById(id, data, options, cb);
  }
});
```
probably can solve the connector failures.